File: DrakborgenDotNet/Hero.cs
using System.Xml.Linq;

namespace Drakborgen
{
    internal class Hero
    {
        internal Card? StatsCard { get; private set; }
        internal int MaxLife { get; init; }
        internal string Id { get; init; }
        internal string Name { get; init; }

        internal Hero(string id, string name, int maxLife)
        {
            Id = id;
	Name = name;

            MaxLife = maxLife;
            StatsCard = new Card(Setup.MediaPath + "hjaltekort/" + id + ".png");
        }

        internal Card PullStatsCard()
        {
	if (StatsCard == null)
	{
	    throw new Exception();
            }
            Card result = StatsCard;
            StatsCard = null;
	return result;
        }

        internal void PlaceStatsCard(Card card)
        {
            if (StatsCard != null)
            {
                throw new Exception();
            }
            StatsCard = card;
        }
    }
}

File: src/Drakborgen/Hero.cpp
#include "Hero.h"
#include "Setup.h"

Hero::Hero(const std::string& id, const std::string& name, int maxLife)
	: id(id)
	, name(name)
	, statsCard(std::make_unique<Card>(Setup::getMediaPath() + "hjaltekort/" + id + ".png"))
	, maxLife(maxLife)
{
}

Hero::Hero(Hero&& other) noexcept
{
	*this = std::move(other);
}

Hero& Hero::operator=(Hero&& other) noexcept
{
	id = std::move(other.id);
	name = std::move(other.name);
	statsCard = std::move(other.statsCard);
	maxLife = other.maxLife;
	return *this;
}

std::unique_ptr<Card> Hero::pullStatsCard()
{
	return std::move(statsCard);
}

void Hero::placeStatsCard(std::unique_ptr<Card> card)
{
	statsCard = std::move(card);
}
